Gems/Atom/RPI/Code/Source/RPI.Public/Pass/ParentPass.cpp

```diff
@@ -71,6 +71,35 @@ namespace AZ
             }
         }
 
+        bool ParentPass::InsertChild(const Ptr<Pass>& child, ChildPassIndex position)
+        {
+            AZ_Assert(child->m_parent == nullptr, "Can't add Pass that already has a parent. Remove the Pass from it's parent before adding it to another Pass.");
```

Comment:
Should that condition return `false` like the next condition, I'm not sure if we can recover from that case.

Reply:
That's a valid point. The condition was copied from AddChild().
The current logic could lead to the same pass was added as child for more than one parent (in their children list).